src/gui/editors/SongEditor.cpp

```diff
@@ -531,35 +535,64 @@ void SongEditor::keyPressEvent( QKeyEvent * ke )
 }
 
 
-
-
 void SongEditor::wheelEvent( QWheelEvent * we )
 {
 	if( we->modifiers() & Qt::ControlModifier )
 	{
 		int z = m_zoomingModel->value();
+		unsigned pos = 0;
+
+		// if Control && Shitf Modifiers, use predefinedZoom values
+		if (we->modifiers() & Qt::ShiftModifier)
+		{
+			int i = 1;
+			do
+			{
+				if (PredefinedZoom[i] >= z)
+				{
+					pos = i;
+					i = PredefinedZoom.size() + 1;
+				}
+
+			} while (i++ < PredefinedZoom.size());
+		}
 
 		if(we->angleDelta().y() > 0)
 		{
-			z++;
+			(we->modifiers() & Qt::ShiftModifier)
+				? z = PredefinedZoom[(pos < (PredefinedZoom.size() - 1)) ? pos + 1 : pos]
+				: z = static_cast<int>(ceil((floor(z / 100.0f * DEFAULT_PIXELS_PER_BAR) + 1) * 100.0f / DEFAULT_PIXELS_PER_BAR));
```

Comment:
```suggestion
			z = (we->modifiers() & Qt::ShiftModifier)
				? PredefinedZoom[std::min(pos + 1, PredefinedZoom.size() - 1)]
				: static_cast<int>(std::ceil((std::floor(z / 100.0f * DEFAULT_PIXELS_PER_BAR) + 1) * 100.0f / DEFAULT_PIXELS_PER_BAR));
```
The ternary operator should only be used to selected one of two values (see [coding conventions](https://github.com/LMMS/lmms/wiki/Coding-Conventions#ternary-operator)). `std::min` can simplify selecting the array index.